### PR TITLE
[battery_plus] Migrate battery code to battery_plus

### PR DESCRIPTION
**Pull request by HakkyuKim**
The only significant change is adding some dart code to set the correct platform interface for Tizen.
Please see `lib/battery_plus_tizen.dart` and `lib/method_channel_battery_plus_tizen.dart` for details.

### CODE REVIEW

File: packages/battery_plus/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+The Tizen implementation of [`battery_plus`](https://github.com/fluttercommunity/plus_plugins/tree/main/packages/battery_plus).
+
+## Usage
+
+This package is not an _endorsed_ implementation of `battery_plus`. Therefore, you have to include `battery_plus_tizen` alongside `battery_plus` as dependencies in your `pubspec.yaml` file.
+
+```yaml
+dependencies:
+  battery_plus: ^1.0.1
+  battery__plus_tizen: ^1.0.0
```
Comment (swift-kim) [inline packages/battery_plus/README.md:12]:
```suggestion
  battery_plus_tizen: ^1.0.0
```

File: packages/battery_plus/LICENSE [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2020 Samsung Electronics Co., Ltd. All rights reserved.
+// Copyright 2017 The Chromium Authors. All rights reserved.
```
Comment (swift-kim) [inline packages/battery_plus/LICENSE:2]:
```suggestion
// Copyright (c) 2017 The Chromium Authors. All rights reserved.
```
For consistency.

File: packages/battery_plus/LICENSE [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//    * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//    * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//    * Neither the name of Google Inc. nor the names of its
```
Comment (swift-kim) [inline packages/battery_plus/LICENSE:14]:
```suggestion
//    * Neither the names of the copyright holders nor the names of the
```

File: packages/battery_plus/lib/battery_plus_tizen.dart [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import 'method_channel_battery_plus_tizen.dart';
+
+/// The Tizen implementation of [BatteryPlatform].
+///
+/// This class sets the default instance of [BatteryPlatform] to
+/// Tizen's implementation of [MethodChannelBattery].
+/// This is required to avoid platform interface being set to Linux's.
+/// https://github.com/fluttercommunity/plus_plugins/blob/06dd9fd986d52ccd72754b2e787198f7c82878b2/packages/battery_plus/lib/battery_plus.dart#L43-L47
+class BatteryPlugin extends BatteryPlatform {
```
Comment (swift-kim) [inline packages/battery_plus/lib/battery_plus_tizen.dart:15]: Why not simply:

```dart
class BatteryPlugin extends MethodChannelBattery {
  /// Registers this class as the default instance of [BatteryPlatform].
  static void register() {
    BatteryPlatform.instance = BatteryPlugin();
  }
}
```

Note that `MethodChannelBattery` is of the platform interface package.

**Review by swift-kim — COMMENTED** [review]
Tip for reviewers: Comparing two files in VS Code is as simple as:

image

File: packages/battery_plus/lib/battery_plus_tizen.dart (hunk 4 above)
Comment (swift-kim) [reply packages/battery_plus/lib/battery_plus_tizen.dart:15]: Ah sorry. I now understand what you mean. Never mind.

File: packages/battery_plus/lib/battery_plus_tizen.dart [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  static void register() {
+    BatteryPlatform.instance = MethodChannelBattery();
+  }
+}
```
Comment (swift-kim) [inline packages/battery_plus/lib/battery_plus_tizen.dart:19]: You can still remove your own `MethodChannelBattery` and simplify the implementation like:

```dart
class BatteryPlugin extends BatteryPlatform {
  /// Registers this class as the default instance of [BatteryPlatform].
  static void register() {
    BatteryPlatform.instance = BatteryPlugin();
  }

  final MethodChannelBattery _methodChannelBattery = MethodChannelBattery();

  @override
  Future<int> get batteryLevel => _methodChannelBattery.batteryLevel;

  @override
  Stream<BatteryState> get onBatteryStateChanged =>
      _methodChannelBattery.onBatteryStateChanged;
}
```

Comment (HakkyuKim) [reply packages/battery_plus/lib/battery_plus_tizen.dart:19]: Brilliant!

File: packages/battery_plus/README.md (hunk 1 above)
Comment (HakkyuKim) [reply packages/battery_plus/README.md]: Thanks for finding the mistake!

**Review by swift-kim — APPROVED** [review]

**Comment by HakkyuKim** [comment]
part of #116

**Pull request merged**